native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/DefaultHttpResponseFuture.java

```diff
@@ -118,6 +119,12 @@ public void notifyHttpListener(HttpCarbonMessage httpCarbonMessage) {
     public void notifyHttpListener(Throwable throwable) {
         responseLock.lock();
         try {
+            if (outboundMsgHolder != null) {
```

Comment:
Do we actually use this constructor if not we don't need the above null check.

https://github.com/ballerina-platform/module-ballerina-http/pull/837/files#diff-3f887598e6e7155d27926134275f68ae74193fc70bc11a3935ee8319206a608cR73

Reply:
Yes, we are using the constructor with `null` outboundMsgHolder, in `notifyListenerAndGetErrorResponseFuture` method and in `HttpCarbonMessage` class.